includes/admin/class-sensei-onboarding.php

```diff
@@ -97,4 +99,59 @@ public function onboarding_page() {
 		<?php
 	}
 
+	/**
+	 * Check if should show help tab or not.
+	 *
+	 * @param string $screen_id Screen ID to check if should show the help tab.
+	 *
+	 * @return boolean
+	 */
+	private function should_show_help_screen( $screen_id ) {
+		return 'edit-course' === $screen_id;
```

Comment:
@donnapep, do you think this filter for the screens to display the tab is good? It displays the tab for the course menu and submenu in `wp-admin`.

Reply:
I was just taking a closer look at this and I'm no longer able to reproduce WooCommerce's Help tab showing on the Courses page when on the Sensei `master` branch with WC activated. Neither can I reproduce when running Sensei LMS `master` + WCPC, or when running SWCPC. So I have no idea how it was possible that I was seeing that Help tab before. Are you still able to reproduce when on Sensei `master` branch?

As the screen ID for the Courses page is `edit-course`, the only place that screen ID could possibly be added by WC is [here](https://github.com/woocommerce/woocommerce/blob/6b61d5f025ba3e77505b43c1c82159ccc3bb9292/includes/admin/wc-admin-functions.php#L44) or [here](https://github.com/woocommerce/woocommerce/blob/6b61d5f025ba3e77505b43c1c82159ccc3bb9292/includes/admin/wc-admin-functions.php#L51), but I didn't drill down any further than that.

I think we first need to figure out how to reliably reproduce this before we decide what to do about it. Let me know if you find a clue. 😄

Reply:
Hi @donnapep ! Yes! I'm still able to reproduce. Actually it's added [here](https://github.com/woocommerce/woocommerce-memberships/blob/018490e7e9c62ef0120afc141cf01a62e7169057/includes/admin/class-wc-memberships-admin.php#L254-L257) (memberships).

I'm wondering if we should rethink about remove the `current_screen` action that adds these tabs instead of changing the `woocommerce_screen_ids` filter. I'm thinking it because if the WooCommerce decides to create a new feature based on this filter, it could not work.

In the current code, I think we are safe, but I'm wondering about updates.

Reply:
In the current WooCommerce code, what we miss out is basically notices and some scripts. What would worry me the most is the [scripts](https://github.com/woocommerce/woocommerce/blob/c0ecb79ba2a6c4eff01f2ae9a59e3b6a9fdd4be1/includes/admin/class-wc-admin-assets.php#L164-L200), but I think these scripts are not used in these pages (`edit-course`).

Reply:
> Yes! I'm still able to reproduce. Actually it's added [here](https://github.com/woocommerce/woocommerce-memberships/blob/018490e7e9c62ef0120afc141cf01a62e7169057/includes/admin/class-wc-memberships-admin.php#L254-L257) (memberships).

Ohhh, nice find. In that case, since memberships is the one adding the help tab, we should use their [`wc_memberships_admin_screen_ids` filter](https://github.com/woocommerce/woocommerce-memberships/blob/018490e7e9c62ef0120afc141cf01a62e7169057/includes/admin/class-wc-memberships-admin.php#L267) (if Memberships is activated) to prevent the tab from showing on Sensei pages, and not the `woocommerce_screen_ids` filter.

Reply:
Hi @donnapep ! Inspecting more, I found a hook called `woocommerce_enable_admin_help_tab` which I believe would be more appropriate to prevent the WooCommerce tabs :)

I prevented it based on `'course' === $_GET['post_type']`. And I kept our new tab based on the `current_screen`, as recommended in the [documentation](https://developer.wordpress.org/reference/classes/wp_screen/add_help_tab/). I wanted to use the same to prevent the WC tabs, but the WC filter happens in the `init` hook, where it's not ready to use yet.

https://github.com/Automattic/sensei/pull/3093/commits/fd45c711dfbe2831df17828a482aa1d14ef2c741

And only a correction about what I said before. Our help tabs are being added in the "Courses -> All Courses" menu only, and not in the submenus as I said.

Reply:
This is looking good. I noticed that WC's help tab is still shown for lessons and messages. I know it's not technically part of this issue, but since it's an easy fix, could we remove it from those pages as well?

Reply:
Sure! https://github.com/Automattic/sensei/pull/3093/commits/417f92ba61eda2016301af3993a0f4a5b4ff7492